src/backend/core/models.py

```diff
@@ -1401,3 +1402,191 @@ def get_dns_record_value(self) -> str:
         """Get the DNS TXT record value for this DKIM key."""
         algorithm_enum = DKIMAlgorithmChoices(self.algorithm)
         return f"v=DKIM1; k={algorithm_enum.label}; p={self.public_key}"
+
+
+class MessageTemplate(BaseModel):
+    """Message template model to store reusable message templates and signatures."""
+
+    name = models.CharField(
+        _("name"),
+        max_length=255,
+        help_text=_(
+            "Name of the template (e.g., 'Standard Reply', 'Out of Office', 'Work Signature')"
+        ),
+    )
+
+    description = models.TextField(
+        _("description"),
+        blank=True,
+        help_text=_("Optional description of when to use this template"),
+    )
+
+    html_body = models.TextField(
+        _("HTML body"),
+        blank=True,
+        help_text=_(
+            "HTML email body template with placeholders (e.g., '<p>Dear {recipient_name},</p>') or signature content"
+        ),
+    )
+
+    text_body = models.TextField(
+        _("text body"),
+        blank=True,
+        help_text=_(
+            "Plain text email body template with placeholders (e.g., 'Dear {recipient_name},') or signature content"
+        ),
+    )
+
+    raw_blob = models.ForeignKey(
+        "Blob",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="message_templates",
+        help_text=_("Reference to the blob containing raw template data"),
+    )
```

Comment:
Currently, it will be hard to create a blob for MessageTemplate because the `Blob` model must be linked to a `Mailbox`.